celery/canvas.py

```diff
@@ -1380,7 +1380,7 @@ def _get_app(self, body=None):
                 tasks = self.tasks.tasks  # is a group
             except AttributeError:
                 tasks = self.tasks
-            if tasks:
+            if len(tasks):
```

Comment:
the reason for this change?

Reply:
using len is quite heavy to know if a list or something is empty as it requires parsing it totaly

Reply:
In Python, calling `len()` on a `list` (or `dict` or `set`) is `O(1)` (see this [StackOverflow](https://stackoverflow.com/questions/12342457/what-is-the-big-o-notation-for-the-len-function-in-python) for more information). It *does not* require iterating the list.

Personally, I find it more readable without the `len()` though. I don't think there's a performance benefit or penalty either way.

Reply:
okok i see